inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/ClickHouseDialect.java

```diff
@@ -129,22 +135,40 @@ public List<LogicalTypeRoot> unsupportedTypes() {
     @Override
     public String getUpdateStatement(
             String tableName, String[] fieldNames, String[] conditionFields) {
+        List<String> conditionFieldList = Arrays.asList(conditionFields);
         String setClause =
                 Arrays.stream(fieldNames)
+                        .filter(fieldName -> !conditionFieldList.contains(fieldName))
                         .map(f -> format("%s = :%s", quoteIdentifier(f), f))
                         .collect(Collectors.joining(", "));
+
         String conditionClause =
                 Arrays.stream(conditionFields)
                         .map(f -> format("%s = :%s", quoteIdentifier(f), f))
                         .collect(Collectors.joining(" AND "));
+        Pair<String, String> databaseAndTableName = getDatabaseAndTableName(tableName);
         return "ALTER TABLE "
-                + quoteIdentifier(tableName)
+                + quoteIdentifier(databaseAndTableName.getLeft())
```

Comment:
if here has bug, maybe delete statement have the same problem?

Reply:
I test delete op, not to found it. But I can add it.

Reply:
> if here has bug, maybe delete statement have the same problem?

@thesumery I test delete is ok. I hava add it for delete.